packages/common/src/project/load.ts

```diff
@@ -3,42 +3,60 @@
 
 import fs from 'fs';
 import path from 'path';
+import {NodeVM, VMScript} from '@subql/x-vm2';
 import {plainToClass} from 'class-transformer';
 import {validateSync} from 'class-validator';
 import yaml from 'js-yaml';
 import {ChainTypes} from './models';
 import {ProjectManifestVersioned, VersionedProjectManifest} from './versioned';
 
-export function loadFromJsonOrYaml(file: string): unknown {
-  const {ext} = path.parse(file);
+export function loadFromFile(filePath: string): unknown {
+  const {base, ext} = path.parse(filePath);
 
-  if (ext !== '.yaml' && ext !== '.yml' && ext !== '.json') {
+  if (ext !== '.yaml' && ext !== '.yml' && ext !== '.json' && ext !== '.js') {
     throw new Error(`Extension ${ext} not supported`);
   }
 
-  const rawContent = fs.readFileSync(file, 'utf-8');
-  return yaml.load(rawContent);
+  if (ext === '.js') {
+    const vm = new NodeVM({
+      console: 'inherit',
+      wasm: false,
+      sandbox: {},
+      require: {
+        external: true,
+        context: 'sandbox',
+      },
+      wrapper: 'commonjs',
+      sourceExtensions: ['js', 'cjs'],
+    });
+
+    const script = new VMScript(`module.exports = require('./${base}');`, filePath);
```

Comment:
```suggestion
    const script = new VMScript(`module.exports.default = require('./${base}').default;`, filePath);
```
to ignore non-default exports.

Reply:
I have `const script = new VMScript(`module.exports = require('${filePath}').default;`, filePath).compile();` and a test which will test non-default exports

Reply:
const script = new VMScript(`module.exports = require('${filePath}').default;`, filePath);
It will ignore non-default exports (have written tests).